src/test/java/dev/openfeature/sdk/OpenFeatureClientTest.java

```diff
@@ -56,36 +56,11 @@ void shouldNotThrowExceptionIfHookHasDifferentTypeArgumentThanHookContext() {
         Mockito.verify(logger, never()).error(anyString(), any(), any());
     }
 
-    @Test
-    void mergeContextTest() {
-        String flag = "feature key";
-        boolean defaultValue = false;
-        String targetingKey = "targeting key";
-        EvaluationContext ctx = new ImmutableContext(targetingKey, new HashMap<>());
-        OpenFeatureAPI api = mock(OpenFeatureAPI.class);
-        FeatureProvider mockProvider = mock(FeatureProvider.class);
-        // this makes it so that true is returned only if the targeting key set at the client level is honored
-        when(mockProvider.getBooleanEvaluation(
-                eq(flag), eq(defaultValue), argThat(
-                        context -> context.getTargetingKey().equals(targetingKey)))).thenReturn(ProviderEvaluation.<Boolean>builder()
-                .value(true).build());
-        when(api.getProvider()).thenReturn(mockProvider);
-        when(api.getProvider(any())).thenReturn(mockProvider);
-
-        MockProviderRepository mockProviderRepository = new MockProviderRepository(mockProvider, true);
-        OpenFeatureClient client = new OpenFeatureClient(mockProviderRepository, api, "name", "version");
-        client.setEvaluationContext(ctx);
-
-        FlagEvaluationDetails<Boolean> result = client.getBooleanDetails(flag, defaultValue);
-
-        assertThat(result.getValue()).isTrue();
-    }
-
```

Comment:
This test was brittle and `multi_layer_context_merges_correctly` tests the same functionality much more rigorously.